internal/pkg/pkg_test.go

```diff
@@ -33,35 +34,146 @@ func TestNewPkg(t *testing.T) {
 	var tests = []struct {
 		name        string
 		inputPath   string
+		workingDir  string
 		uniquePath  string
 		displayPath string
 	}{
 		{
 			name:        "test1",
 			inputPath:   ".",
-			displayPath: ".",
+			workingDir:  "foo",
+			displayPath: "foo",
 		},
 		{
 			name:        "test2",
 			inputPath:   "../",
-			displayPath: "..",
+			workingDir:  "foo/bar",
+			displayPath: "foo",
 		},
 		{
 			name:        "test3",
-			inputPath:   "./foo/bar/",
-			displayPath: "foo/bar",
+			inputPath:   "./bar/baz",
+			workingDir:  "foo",
+			displayPath: "baz",
+		},
+		{
+			name:        "test4",
+			inputPath:   "../../foo/bar/baz",
+			workingDir:  "foo/bar",
+			displayPath: "baz",
+		},
+		{
+			name:        "test5",
+			inputPath:   "../../",
+			workingDir:  "foo/bar/baz",
+			displayPath: "foo",
 		},
 	}
 	for i := range tests {
 		test := tests[i]
 		t.Run(test.name, func(t *testing.T) {
+			dir, err := ioutil.TempDir("", "")
+			defer os.RemoveAll(dir)
+			assert.NoError(t, err)
+			err = os.MkdirAll(filepath.Join(dir, "foo", "bar", "baz"), 0700)
+			assert.NoError(t, err)
+			err = os.Chdir(filepath.Join(dir, test.workingDir))
+			assert.NoError(t, err)
 			p, err := New(test.inputPath)
 			assert.NoError(t, err)
 			assert.Equal(t, test.displayPath, string(p.DisplayPath))
 		})
 	}
 }
 
+func TestAdjustDisplayPathForSubpkg(t *testing.T) {
+	var tests = []struct {
+		name        string
+		workingDir  string
+		parentPath  string
+		subPkgPath  string
+		rootPkgPath string
+		displayPath string
+	}{
+		{
+			name:        "test1",
```

Comment:
These names are not helpful to understand what is going on in each testcase..

Reply:
Updated.